File: Data Structures & Algorithms/jump-game-ii/submission-0.cs
public class Solution {
    public int Jump(int[] nums) {
        int n = nums.Length;
       
        if(n <= 1) {
            return 0;
        }
        int l = 0, r = 0, minJumps = 0;
        while(r < n - 1) {
            int farthest = 0;
            for(int i = l; i<=r; i++) {
                farthest = Math.Max(farthest, nums[i] + i);
            }
            l = r + 1;
            r = farthest;
            minJumps++;
        }
        return minJumps;
    }
}

File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int depth = 0;
    public int find(TreeNode root, int depth){
        if(root == null) {
            return 0;
        }
        int leftDepth = Math.max(depth, find(root.left, depth + 1));
        int rightDepth = Math.max(depth, find(root.right, depth + 1));
        return Math.max(leftDepth, rightDepth);
    }
    public int maxDepth(TreeNode root) {
       if(root == null) {
            return 0;
       }
 
       return       find(root, 1);
    }
}
